### ISSUE TITLE
Extend GenericShiftedDataset with stride and max_samples_per_ts functionalities

### ISSUE THREAD

**Issue by maxkazmsft** [unit8co/darts#940]
**Is your feature request related to a current problem? Please describe.**
Right now `GenericShiftedDataset` only strides the data by 1 time step - in some applications, I might want to increase this stride to discard some data (if I have too much data, or if adjacent time steps don't add too much value). This will move the sliding window by more than 1 time step, set by `stride` integer parameter.

Also, `max_samples_per_ts` can be extended to optionally take in a list of integers, which specifies max number of samples for each time series, in case we want to sample a specific amount of points for each time series.

Using the logic above, one can further extend `stride` to also take in a list of integers, which would specify a different stride for each time series.

In summary, extend `max_samples_per_ts` to `Optional[Union[int, Sequence[int]]]` and add `stride` parameter with similar `Optional[Union[int, Sequence[int]]]` for `GenericShiftedDataset`.

**Describe proposed solution**
I think the above should be sufficient - I'm asking to extend the functionality of `GenericShiftedDataset` class.

**Describe potential alternatives**
Right now I'd have to implement my own `TrainingDataset` and `InferenceDataset` to achieve this custom sampling functionality.

**Additional context**
None

**Comment by hrzn** [comment]
Yes, this would definitely provide some value. Please let us know if you would be ready to contribute such a dataset.

**Comment by maxkazmsft** [comment]
That sounds great - what are the steps for contributing? Should I just open a PR from my fork to the main repo?

**Comment by hrzn** [comment]
Yes 👍

**Comment by blacksnail789521** [comment]
> **Describe potential alternatives**
> Right now I'd have to implement my own `TrainingDataset` and `InferenceDataset` to achieve this custom sampling functionality.

Can you share your code of `TrainingDataset` that had implemented `stride`, please?
Also, I think overall it should be more user-friendly to implement `stride` in `model.fit`.

**Comment by hrzn** [comment]
Hi @maxkazmsft, is this still on the table?
